pkg/jobs/registry.go

```diff
@@ -402,16 +411,53 @@ func createJobsInBatchWithTxn(
 		return nil, err
 	}
 
+	// Insert the job payload and details into the system.jobs_info table if the
+	// associated cluster version is active.
+	//
+	// TODO(adityamaru): Stop writing the payload and details to the system.jobs
+	// table once we are outside the compatability window for 22.2.
+	if r.settings.Version.IsActive(ctx, clusterversion.V23_1CreateSystemJobInfoTable) {
+		if err := batchJobWriteToJobInfo(ctx, txn, jobs, modifiedMicros); err != nil {
+			return nil, err
+		}
+	}
+
 	return jobIDs, nil
 }
 
+func batchJobWriteToJobInfo(
+	ctx context.Context, txn isql.Txn, jobs []*Job, modifiedMicros int64,
+) error {
+	marshalPanic := func(m protoutil.Message) []byte {
+		data, err := protoutil.Marshal(m)
+		if err != nil {
+			panic(err)
```

Comment:
we have an error return don't we? why panic?

Reply:
copy pasta, fixed.